cel/README.md

```diff
@@ -37,231 +45,351 @@ kubectl apply --filename https://storage.cloud.google.com/tekton-releases-nightl
 
 ```yaml
+apiVersion: custom.tekton.dev/v1alpha1
+kind: CEL
+metadata:
+  name: get-red
+spec:
+  expressions:
+    - name: expression
+      value: "{'blue': '0x000080', 'red': '0xFF0000'}['red']"
+---
 apiVersion: tekton.dev/v1beta1
 kind: Pipeline
 metadata:
-  generateName: pipeline-
+  generateName: pipeline-get-red-
 spec:
   tasks:
     - name: get-red
       taskRef:
-        apiVersion: cel.tekton.dev/v1alpha1
+        apiVersion: custom.tekton.dev/v1alpha1
         kind: CEL
-      params:
-        - name: red
-          value: "{'blue': '0x000080', 'red': '0xFF0000'}['red']"
+        name: get-red
 ```
-### Configuring a `CEL` `Custom Task` in a `PipelineRun`
+### Configuring a `CEL` in a `PipelineRun`
 
-The `CEL` `Custom Task` can be specified within a `PipelineRun`, as such:
+A `CEL` can be specified within a `PipelineRun`, as such:
 
 ```yaml
+apiVersion: custom.tekton.dev/v1alpha1
+kind: CEL
+metadata:
+  name: get-blue
+spec:
+  expressions:
+    - name: expression
+      value: "{'blue': '0x000080', 'red': '0xFF0000'}['blue']"
+---
 apiVersion: tekton.dev/v1beta1
 kind: PipelineRun
 metadata:
-  generateName: pipelinerun-
+  generateName: pipelinerun-get-blue-
 spec:
   pipelineSpec:
     tasks:
       - name: get-blue
         taskRef:
-          apiVersion: cel.tekton.dev/v1alpha1
+          apiVersion: custom.tekton.dev/v1alpha1
           kind: CEL
-        params:
-          - name: blue
-            value: "{'blue': '0x000080', 'red': '0xFF0000'}['blue']"
+          name: get-blue 
 ```
 
 ### Specifying CEL expressions
 
-The CEL expressions to be evaluated by the `Run` are specified using parameters. The parameters can be specified
-in the `Run` directly or be passed through from a `Pipeline` or `PipelineRun`, as such:
+The CEL expressions to be evaluated by the `CEL` are specified in `expressions` field. 
+
+They are specified as `Parameters`, that is, they are made up of `name` and `value` pairs which are Strings.
+
+```yaml
+apiVersion: custom.tekton.dev/v1alpha1
+kind: CEL
+metadata:
+  name: get-type
+spec:
+  expressions:
+    - name: type
+      value: "type(1)"
+---
+apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+metadata:
+  generateName: pipeline-run-get-type-
+spec:
+  serviceAccountName: 'default'
+  pipelineSpec:
+    tasks:
+      - name: get-type
+        taskRef:
+          apiVersion: custom.tekton.dev/v1alpha1
+          kind: CEL
+          name: get-type
+      - name: echo-get-type
+        when:
+          - input: "$(tasks.get-type.results.type)"
+            operator: in
+            values: ["int"]
+        taskSpec:
+          steps:
+            - name: echo
+              image: ubuntu
+              script: echo ISINT!
+```
+
+For more information about `Parameters`, read [specifying `Parameters`][specifying-parameters].
+
+### Specifying CEL environment variables
+
+The CEL variables to be declared in the environment before evaluation by `CEL` are specified in `variables` field.
```

Comment:
is it possible to explain more here about what CEL environment variables are and why these need to be treated differently from other variables/parameters?

(i think it might be that im just not following the example)

Reply:
yes yes, makes sense -- added a description of the evaluation environment and how you can bind variables to it (with reference to the CEL documentation that goes into further detail